File: frontend/src/Services/UserService.js
import ApiService from "./BaseService/ApiService";

const USER_URL  = '/user'
class UserService {

    getUsers(page, size){ 
        let url = '/users?page='+page+'&size='+size;
        return ApiService.get(USER_URL+url);
    }

    get(url) { 
        return ApiService.get(url)
    }
    getUserByUsername(username){
        return ApiService.get(USER_URL+'/'+username)
    }
    post(data) { 
        return ApiService.post(USER_URL,data)
    }
    update(username, body) {
        return ApiService.put(`/user/${username}`, body); 
    }
    createUserWithRoles(data) {
        return ApiService.post('/user/create', data);
    }

    createMaster(data) {
        return ApiService.post('/user/masters', data);
      }

    getBrigadeMasters(brigadeId) {
        return ApiService.get(`/orders/brigade/${brigadeId}/masters`);
      }
      
    loadImage(username,body) { 
        return ApiService.put(USER_URL+"/upload-image/"+username,body)
    }
    getQualifications() {
    
        return ApiService.get('/user/qualifications');
      }
    deleteUserById = (id, token) => {
        return fetch(`http://localhost:8501/api/user/${id}`, {
            method: 'DELETE',
            headers: {
                'Authorization': `Bearer ${token}`
            }
      
        
        }).then(async response => {
            const contentType = response.headers.get("content-type");
            if (!response.ok) {
                if (contentType && contentType.includes("application/json")) {
                    const data = await response.json();
                    throw data;
                } else {
                    throw new Error("Unexpected server response");
                }
            }
            return response.json();
        });
    };

    restoreUser = (id, token) => {
        if (!token) {
            throw new Error('Authorization token is missing');
        }
        return fetch(`http://localhost:8501/api/user/${id}/restore`, {
            method: 'PUT',
            headers: {
                'Authorization': `Bearer ${token}`
            }
        }).then(async (response) => {
            if (!response.ok) {
                const errorText = await response.text();
                console.error('Restore user failed:', errorText);
                throw new Error('Failed to restore user');
            }
            return response.json();
        });
    };
    

    getUserNotifications() {
        return ApiService.get('/notifications');
      }
    //put(url, data) { return axios.put(API_BASE_URL + url, data); }

    //delete(url) { return axios.delete(API_BASE_URL + url); }
}

export default new UserService();

File: admin_flutter/lib/services/user_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'auth_service.dart';

class UserService {
  static const String baseUrl = 'http://172.26.205.54:8501/api';
  final AuthService _authService = AuthService();
  
  Future<Map<String, String>> _getHeaders() async {
    final token = await _authService.getToken();
    return {
      'Content-Type': 'application/json',
      'Authorization': 'Bearer ${token ?? ''}',
    };
  }
  
  Future<Map<String, dynamic>> getUsers(int page, int size) async {
    try {
      final headers = await _getHeaders();
      final response = await http.get(
        Uri.parse('$baseUrl/user/users?page=$page&size=$size'),
        headers: headers,
      );
      
      if (response.statusCode == 200) {
        return jsonDecode(response.body);
      } else {
        throw Exception('Ошибка загрузки пользователей');
      }
    } catch (e) {
      throw Exception(e.toString());
    }
  }
  
  Future<Map<String, dynamic>> getUserByUsername(String username) async {
    try {
      final headers = await _getHeaders();
      final response = await http.get(
        Uri.parse('$baseUrl/user/$username'),
        headers: headers,
      );
      
      if (response.statusCode == 200) {
        return jsonDecode(response.body);
      } else {
        throw Exception('Пользователь не найден');
      }
    } catch (e) {
      throw Exception(e.toString());
    }
  }
  
  Future<Map<String, dynamic>> updateUser(String username, Map<String, dynamic> data) async {
    try {
      final headers = await _getHeaders();
      final response = await http.put(
        Uri.parse('$baseUrl/user/$username'),
        headers: headers,
        body: jsonEncode(data),
      );
      
      if (response.statusCode == 200) {
        return jsonDecode(response.body);
      } else {
        final error = jsonDecode(response.body);
        throw Exception(error['message'] ?? 'Ошибка обновления пользователя');
      }
    } catch (e) {
      throw Exception(e.toString());
    }
  }
  
  Future<Map<String, dynamic>> createUser(Map<String, dynamic> data) async {
    try {
      final headers = await _getHeaders();
      final response = await http.post(
        Uri.parse('$baseUrl/user/create'),
        headers: headers,
        body: jsonEncode(data),
      );
      
      if (response.statusCode == 200 || response.statusCode == 201) {
        return jsonDecode(response.body);
      } else {
        final error = jsonDecode(response.body);
        throw Exception(error['message'] ?? 'Ошибка создания пользователя');
      }
    } catch (e) {
      throw Exception(e.toString());
    }
  }
  
  Future<void> deleteUser(int userId) async {
    try {
      final headers = await _getHeaders();
      final response = await http.delete(
        Uri.parse('$baseUrl/user/$userId'),
        headers: headers,
      );
      
      if (response.statusCode != 200) {
        throw Exception('Ошибка удаления пользователя');
      }
    } catch (e) {
      throw Exception(e.toString());
    }
  }
  
  Future<void> restoreUser(int userId) async {
    try {
      final headers = await _getHeaders();
      final response = await http.put(
        Uri.parse('$baseUrl/user/$userId/restore'),
        headers: headers,
      );
      
      if (response.statusCode != 200) {
        throw Exception('Ошибка восстановления пользователя');
      }
    } catch (e) {
      throw Exception(e.toString());
    }
  }
}


